src/Orleans/Messaging/Message.cs

```diff
@@ -601,7 +606,7 @@ private List<ArraySegment<byte>> Serialize_Impl(out int headerLengthOut, out int
             {
                 BufferPool.GlobalPool.Release(headerBytes);
             }
-            headerBytes = headerStream.ToBytes() as List<ArraySegment<byte>>;
+            headerBytes = context.Stream.ToBytes() as List<ArraySegment<byte>>;
             int headerLength = headerBytes.Sum(ab => ab.Count);
```

Comment:
Yes I saw that too, but I was able to overcome that restriction in an "else" branch above, but sure we can exclude it in this PR, not a blocking for it and I saw another place in Message.cs which can be enhanced.